packages/rum-core/src/domain/profiling/exporter/exportToJsonIntake.ts

```diff
@@ -0,0 +1,165 @@
+    RumProfilerTraceExporter,
+    RumProfilerTrace,
+} from '../types';
+import { getLongTaskId } from '../utils/longTaskRegistry';
+
+import { getEmptyPprofBlob } from './emptyPprof';
+
+interface ProfileEventAttributes {
+    application: { id: string };
+    session?: { id: string };
+    view?: { name: string[] };
+    context?: { profile_long_task_id: string[] };
+}
+interface ProfileEvent extends ProfileEventAttributes{
+    attachments: string[];
+    start: string // ISO date
+    end:  string // ISO date
+    family: 'chrome';
+    tags_profiler: string;
+}
+
+/**
+ * Exports RUM profile as JSON to public profiling intake.
+ */
+export const exportToJSONIntake: RumProfilerTraceExporter = (
+    profilerTrace,
+    endpointBuilder,
+    applicationId,
+    sessionId,
+) => {
+    const event = createProfileEvent(profilerTrace, endpointBuilder, applicationId, sessionId);
+    const payload = createProfilePayload(profilerTrace, event)
+
+    // Sends profile to public profiling intake.
+    const xhrUrl = endpointBuilder.build('xhr', payload)
+
+    // Send everything.
+    return new Promise((resolve) =>  sendXHR(xhrUrl, payload.data, resolve));
+};
+
+function createProfileEvent(
+    profilerTrace: RumProfilerTrace, 
+    endpointBuilder: EndpointBuilder, 
+    applicationId: string ,
+    sessionId: string | undefined
+): ProfileEvent {
+    const rawParameters = endpointBuilder.rawParameters;
+    const profileAttributes = extractProfileEventAttributes(profilerTrace, applicationId, sessionId);
+    const profileEventTags = extractProfileEventTags(rawParameters.configurationTags, rawParameters.initConfiguration)
+
+    const start = new Date(profilerTrace.timeOrigin + profilerTrace.startTime)
+    const end = new Date(profilerTrace.timeOrigin + profilerTrace.endTime)
+    
+    const profileEvent: ProfileEvent = {
+        ...profileAttributes,
+        attachments: ['wall-time.pprof', 'wall-time.json'],
+        start: start.toISOString(),
+        end: end.toISOString(),
+        family: 'chrome',
+        tags_profiler: profileEventTags.join(','),
+    };
+
+    return profileEvent;
+}
+
+function extractProfileEventTags(configurationTags: string[], initConfiguration: InitConfiguration): string[] {
+
+    const profileEventTags = configurationTags.concat([
+        `service:${initConfiguration.service}`,
+        `version:${initConfiguration.version}`,
+        `env:${initConfiguration.env || 'unknown'}`,
+        'language:javascript',
+        'runtime:chrome',
+        'family:chrome',
+        'format:json',
+        // TODO: replace with RUM device id in the future
+        `host:${normalizeTag(navigator.userAgent).slice(0, 200)}`,
+    ]);
+
+    // TODO deobfuscation is not supported yet
+    // if (config.commitHash) {
+    //     eventTags.push(`git.commit.sha:${config.commitHash}`);
+    // }
+    // if (config.repositoryUrl) {
+    //     eventTags.push(`git.repository_url:${config.repositoryUrl}`);
+    // }
```

Comment:
This is something we need to figure out, where to do this de-obfuscation. This code will be removed before merging obviously.